tests/test_utils.py

```diff
@@ -299,3 +301,63 @@ def check(data, data_required_keys, data_optional_keys, result):
         ({'key1': 'value1'}, [], ['key2'], False),
     ]:
         check(data, data_required_keys, data_optional_keys, result)
+
+
+class TestSqlCompareDateString:
+
+    class Timestamp(datetime.datetime):
+        "An object that looks like an SQL TIMESTAMP value"
+
+        def between(self, a, b):
+            return a <= self and self <= b
+
+    def test_timestamp(self):
+        assert self.Timestamp(2000, 1, 1)
+        assert self.Timestamp(2000, 1, 1) == datetime.datetime(2000, 1, 1)
+        assert self.Timestamp(2000, 1, 1) >= datetime.datetime(2000, 1, 1)
+        assert self.Timestamp(2000, 1, 1) <= datetime.datetime(2000, 1, 1)
+        assert self.Timestamp(2000, 1, 1) > datetime.datetime(1999, 1, 1)
+        assert self.Timestamp(2000, 1, 1) < datetime.datetime(2001, 1, 1)
+        assert self.Timestamp(2000, 1, 1, 12) >= datetime.datetime(2000, 1, 1)
+        assert self.Timestamp(2000, 1, 1, 12) <= datetime.datetime(2000, 1, 2)
+        assert self.Timestamp(2000, 1, 1, 12).between(datetime.datetime(2000, 1, 1), datetime.datetime(2000, 1, 2))
+
+    def test_can_test_whether_timestamp_is_during_day(self):
+        timestamp = self.Timestamp(2010, 12, 2, 15, 34)
+        assert sql_compare_date_string(timestamp, "2010-12-02")
```

Comment:
I think the function name gets confusing when it's just a straight 'is it this date, yes/no'. How about `compare_sql_datetime_with_date_string`?

Reply:
Yes, the name is definitely the thing I spent the longest on...

I like your suggestion, what about (along the same lines) `compare_sql_datetime_with_string`?

Reply:
Yep - let's go with that 👍

Reply:
Done